tests/unit/controller/PageControllerTest.php

```diff
@@ -73,7 +73,8 @@ public function testIndex() {
 		$this->mockUrlToUploadEndpoint($url);
 		$params = [
 			'appName' => $this->appName,
-			'uploadUrl' => $url
+			'uploadUrl' => $url,
+			'publicUploadEnabled' => $this->appConfig->getAppValue('core', 'shareapi_allow_public_upload', 'yes')
```

Comment:
Please mock the method, otherwise your test depends on your admin settings.
See: https://github.com/imjalpreet/gallery/blob/Issue565Test/tests/unit/controller/PageControllerTest.php#L210-L219

It would probably be best to return a map. Let me know if you need a pointer, but there are examples available in the unit test classes.